apple/internal/ios_rules.bzl

```diff
@@ -1113,6 +1113,18 @@ def _ios_dynamic_framework_impl(ctx):
 
     providers = processor_result.providers
 
+    new_providers = []
+    for provider in providers:
+        new_providers.append(provider)
+        if type(provider) == "AppleDynamicFramework":
+            # Make the ObjC provider using the framework_files depset found
+            # in the AppleDynamicFramework provider. This is to make the
+            # ios_dynamic_framework usable as a dependency in swift_library
+            objc_provider_fields = {}
+            objc_provider_fields["dynamic_framework_file"] = provider.framework_files
+            objc_provider = apple_common.new_objc_provider(**objc_provider_fields)
```

Comment:
Actually, since you are only passing one field, always, can you instead call `new_objc_provider` with the one field?